lib/chef/provider/package/snap.rb

```diff
@@ -268,51 +283,61 @@ def generate_multipart_form_data(snap_name, action, options, path, content_lengt
 
         # Constructs json to post for snap changes
         #
-        #   @param snap_names [Array] An array of snap package names to install
         #   @param action [String] The action.  install, refresh, remove, revert, enable, disable or switch
         #   @param channel [String] The release channel.  Ex. stable
-        #   @param options [Hash] Misc configuration Options
+        #   @param options [Array] An array of configuration Options
+        #   @param version [String] A snap package version
         #   @param revision [String] A revision/version
-        def generate_snap_json(snap_names, action, channel, options, revision = nil)
+        def generate_snap_json(action, channel, options, version, revision = nil)
           request = {
               "action" => action,
-              "snaps" => snap_names,
```

Comment:
So the problem with this is that if snap grows the concept of transactions that this change will break transactions (and this is something to be very worried about because package managers will inevitably grow this concern if snap doesn't have it already).  So when we're given multiple packages we should install them all together as a batch.

Since it looks like the API doesn't support installing individual versions when installing multiple packages that means we can't either, but we shouldn't shape the whole API around that edge case.  We should just throw if the user attempts to do a multipackage install with pinned versions and state that snap does not support that (yet?).  Then when given a single package, with a version, we can do the individual install of the pinned package.  That also makes it obvious to the end user that we cannot do a single-transaction install of multiple pinned packages, forces them to do a series of single snap_package calls, which is entirely the correct way for them to think about it.